feature-libs/order/components/order-details/order-details.module.ts

```diff
@@ -114,6 +160,19 @@ const moduleComponents = [
     }),
     provideDefaultConfig(defaultConsignmentTrackingLayoutConfig),
     provideDefaultConfig(defaultReorderLayoutConfig),
+    provideDefaultConfigFactory(() =>
+      inject(MYACCOUNT_ORDER_ENHANCED_UI) ? enhancedUICmsMapping : {}
+    ),
+    {
+      provide: MODULE_INITIALIZER,
+      useFactory: registerOrderOutletFactory,
+      deps: [
+        MYACCOUNT_ORDER_ENHANCED_UI,
+        OutletService,
+        ComponentFactoryResolver,
+      ],
```

Comment:
[MEDIUM]
Declaring `deps` array is no longer recommended, because we can use `inject()` function inside the factory function (and add/remove dependencies without breaking changes):
```suggestion
```
+
```ts
function registerOrderOutletFactory(): () => void {
  const token = inject(MYACCOUNT_ORDER_ENHANCED_UI);
  const outletService = inject(OutletService);
  const componentFactoryResolver = inject(ComponentFactoryResolver);

  return () => {
    /*...*/
  };
```
Note: `inject()` function should be invoked at the top level of the factory function's body. In particular it shouldn't be invoked in nested function (the returned module initializer callback) - becasue then it won't work. For more, see docs for [Usage notes of inject function](https://angular.io/api/core/inject#usage-notes)

Reply:
done